src/components/input-panel/compiled-spec-header/index.tsx

```diff
@@ -20,6 +21,37 @@ class CompiledSpecDisplayHeader extends React.PureComponent<Props> {
     super(props);
     this.editVegaSpec = this.editVegaSpec.bind(this);
   }
+
+  public handleMergeConfig(e) {
+    e.stopPropagation();
+
+    // check if the configEditorStringisEmpty
+    if (this.props.configEditorString === '{}') {
+      return;
+    }
+
+    // merge config into spec
+    try {
+      const spec = JSON.parse(this.props.editorString);
+      const config = JSON.parse(this.props.configEditorString);
+      if (spec.config) {
+        spec.config = mergeDeep(config, spec.config);
+      } else {
+        spec.config = config;
+      }
+      this.props.updateEditorString(stringify(spec));
+
+      // clear config
+      this.props.clearConfig();
+
+      // run the config if auto mode is selected
+      if (!this.props.manualParse) {
+        this.props.parseSpec(true);
+      }
+    } catch (e) {
+      console.warn(e);
```

Comment:
what is this? Why hide this from users?

Reply:
The user already gets the error in the auto mode.
In manual mode , we can run the config which will again give the error , so no need of extra error handling here.